lib/vmpooler/pool_manager.rb

```diff
@@ -21,6 +21,9 @@ def initialize(config, logger, redis, metrics)
 
       # Our thread-tracker object
       $threads = {}
+
+      # Host tracking object
+      $provider_hosts = {}
```

Comment:
nit: REALLY REALLY try to avoid adding more global variables.  Instance level should be enough.

Reply:
@glennsarti are you sure? I specifically require that this variable be accessible across all vmpooler provider threads.